perf/reader_test.go

```diff
@@ -289,6 +290,18 @@ func TestCreatePerfEvent(t *testing.T) {
 		t.Fatal("Can't create perf event:", err)
 	}
 	unix.Close(fd)
+
+	onlineCPU, err := internal.OnlineCPUs()
+	if err != nil {
+		t.Fatal("perf event array: ", err)
+	}
+	fd, err = createPerfEvent(onlineCPU, 1)
+	if !xerrors.Is(err, syscall.ENODEV) {
```

Comment:
If you check the Linux sources you'll find that ENODEV is only returned if:
- onlineCPU is [less than nr_cpu_ids](https://elixir.bootlin.com/linux/v5.5.9/source/kernel/events/core.c#L10685)
- and the [CPU is offline](https://elixir.bootlin.com/linux/v5.5.9/source/kernel/events/core.c#L11472)

I think you're violating the first condition, and getting EINVAL therefore.